docs/tech-notes/life_of_a_query.md

```diff
@@ -882,40 +900,38 @@ Let's go through the code a bit:
-   are present in the [(one and only) meta1-range](https://github.com/cockroachdb/cockroach/blob/d5f73c6a05c99f7c6b40b80dea80e8ed8ab15cac/pkg/kv/kvclient/rangecache/range_cache.go#L791),
+   a meta2-range (a second-level index range), and the descriptors for meta2-ranges
+   are present in the (one and only) [meta1-range](https://github.com/cockroachdb/cockroach/blob/d5f73c6a05c99f7c6b40b80dea80e8ed8ab15cac/pkg/kv/kvclient/rangecache/range_cache.go#L791),
    which is [provided](https://github.com/cockroachdb/cockroach/blob/4e4f31a0ed1a8ea985b9ab6f72e29266b259900e/pkg/server/server.go#L404) by [`Gossip.GetFirstRangeDescriptor()`](https://github.com/cockroachdb/cockroach/blob/8c5253bbccea8966fe20e3cb8cc6111712f6f509/pkg/gossip/gossip.go#L1596).
    
 2. Each sub-request (partial batch) is sent to its range. This is done
-   through the call to
+   through a call to
    [`DistSender.sendPartialBatchAsync()`](https://github.com/cockroachdb/cockroach/blob/904807640f6a28aa82c5e63f0b62de2dbf27b3a0/pkg/kv/kvclient/kvcoord/dist_sender.go#L1259)
    which
    [truncates](https://github.com/cockroachdb/cockroach/blob/904807640f6a28aa82c5e63f0b62de2dbf27b3a0/pkg/kv/kvclient/kvcoord/dist_sender.go#L1416)
-   all the requests in the batch to the current range and then it
+   all the requests in the batch to the current range and then
    [sends](https://github.com/cockroachdb/cockroach/blob/904807640f6a28aa82c5e63f0b62de2dbf27b3a0/pkg/kv/kvclient/kvcoord/dist_sender.go#L1478)
-   the truncated batch to a range. All these partial batches are sent
-   concurrently.
+   the truncated batch to it. All these partial batches are sent concurrently.
 
-   `sendPartialBatch()` is the level at which an error stemming from stale
-   `RangeCache` information is handled: the range descriptor that's detected to be stale is
+   `sendPartialBatch()` is where errors stemming from stale `RangeCache`
+   information is handled: the stale range descriptor is
    [evicted](https://github.com/cockroachdb/cockroach/blob/904807640f6a28aa82c5e63f0b62de2dbf27b3a0/pkg/kv/kvclient/kvcoord/dist_sender.go#L1523)
    from the cache and the partial batch is retried.
 
-3. Sending a partial batch to a single range implies selecting the
-   right replica of that range and performing an RPC to it. By default,
-   each range is replicated three ways, but only one of the three
-   replicas is the "lease holder" - the temporarily designed owner of
-   that range, in charge of coordinating all reads and writes to it (see
-   the [Range Leases
+3. Sending a partial batch to a single range involves selecting the appropriate
+   replica for that range and making an RPC call to it. By default, each range is
+   replicated three ways, but only one of the three replicas is the "lease holder" —
+   the temporarily designated owner of that range, in charge of coordinating all
+   reads and writes to it (see the [Range Leases
```

Comment:
"all reads" isn't entirely correct due to follower reads, but maybe that's ok to gloss over here.

Reply:
Follower reads are out of scope here. We say in the introduction that we're omitting a bunch of special cases.